Kernel/FileSystem/DevLoopFS/FileSystem.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2024, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Devices/DeviceManagement.h>
+#include <Kernel/Devices/Loop/LoopDevice.h>
+#include <Kernel/FileSystem/DevLoopFS/FileSystem.h>
+#include <Kernel/FileSystem/DevLoopFS/Inode.h>
+#include <Kernel/FileSystem/RAMBackedFileType.h>
+#include <Kernel/Time/TimeManagement.h>
+
+namespace Kernel {
+
+ErrorOr<NonnullRefPtr<FileSystem>> DevLoopFS::try_create(ReadonlyBytes)
+{
+    return TRY(adopt_nonnull_ref_or_enomem(new (nothrow) DevLoopFS));
+}
+
+DevLoopFS::DevLoopFS() = default;
+DevLoopFS::~DevLoopFS() = default;
+
+u8 DevLoopFS::internal_file_type_to_directory_entry_type(DirectoryEntryView const& entry) const
+{
+    return ram_backed_file_type_to_directory_entry_type(entry);
+}
+
+ErrorOr<void> DevLoopFS::initialize()
+{
+    m_root_inode = TRY(adopt_nonnull_ref_or_enomem(new (nothrow) DevLoopFSInode(*this, 1, nullptr)));
+    m_root_inode->m_metadata.inode = { fsid(), 1 };
+    m_root_inode->m_metadata.mode = 0040555;
```

Comment:
hrm. I'd like to avoid raw octal constants anywhere in the code. that sounds like a separate cleanup refactor PR though.

Reply:
I'll try to take care of this soon.